### PR TITLE
Port1.10/storage rendering

### PR DESCRIPTION
**Pull request by Raycoms**
Ports possible storage upgrade and improved rendering to 1.10.

# Changes proposed in this pull request:
-
-
-

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/structures/helpers/Structure.java [hunk 1]
```diff
@@ -433,11 +457,17 @@ public boolean isCorrectMD5(final String otherMD5)
     /**
      * Reads a template from an inputstream.
      */
-    private static Template readTemplateFromStream(final InputStream stream) throws IOException
+    private static Template readTemplateFromStream(final InputStream stream, final DataFixer fixer) throws IOException
     {
         final NBTTagCompound nbttagcompound = CompressedStreamTools.readCompressed(stream);
+
+        if (!nbttagcompound.hasKey("DataVersion", 99))
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/structures/helpers/Structure.java:464]: MAJOR Assign this magic number 99 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/structures/helpers/Structure.java [hunk 2]
```diff
@@ -433,11 +457,17 @@ public boolean isCorrectMD5(final String otherMD5)
...
+        {
+            nbttagcompound.setInteger("DataVersion", 500);
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/structures/helpers/Structure.java:466]: MAJOR Assign this magic number 500 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/structures/helpers/Structure.java [hunk 3]
```diff
@@ -508,42 +538,71 @@ public void setPlacementSettings(final PlacementSettings settings)
...
+                }
+                fakeWorld.setTileEntity(aBlockList.pos, tileentity);
             }
 
-            final BlockPos blockpos = aBlockList.pos.add(startingPos);
-            final IBlockState iBlockExtendedState = block.getExtendedState(iblockstate, clientWorld, blockpos);
-            final IBakedModel ibakedmodel = Minecraft.getMinecraft().getBlockRendererDispatcher().getModelForState(iblockstate);
-            TileEntity tileentity = null;
-            if (block.hasTileEntity(iblockstate) && aBlockList.tileentityData != null)
+            for (final Template.BlockInfo aBlockList : blockList)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/structures/helpers/Structure.java:570]: MINOR Combine this loop with the one that starts on line 556. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3047)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 40 issues
* ![BLOCKER][BLOCKER] 3 blocker
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 34 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [Structure.java#L137](https://github.com/Minecolonies/minecolonies/blob/8e160f2003a1e8f657924b5bc00191bb0ce9fa03/src/main/java/com/minecolonies/structures/helpers/Structure.java#L137): Close this "InputStream" in a "finally" clause. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2095)
1. ![BLOCKER][BLOCKER] [Structure.java#L247](https://github.com/Minecolonies/minecolonies/blob/8e160f2003a1e8f657924b5bc00191bb0ce9fa03/src/main/java/com/minecolonies/structures/helpers/Structure.java#L247): Close this "InputStream" in a "finally" clause. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2095)
1. ![BLOCKER][BLOCKER] [Structure.java#L260](https://github.com/Minecolonies/minecolonies/blob/8e160f2003a1e8f657924b5bc00191bb0ce9fa03/src/main/java/com/minecolonies/structures/helpers/Structure.java#L260): Close this "InputStream" in a "finally" clause. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2095)
1. ![MAJOR][MAJOR] [Structure.java#L63](https://github.com/Minecolonies/minecolonies/blob/8e160f2003a1e8f657924b5bc00191bb0ce9fa03/src/main/java/com/minecolonies/structures/helpers/Structure.java#L63): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 37 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/8e160f2003a1e8f657924b5bc00191bb0ce9fa03/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/8e160f2003a1e8f657924b5bc00191bb0ce9fa03/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/8e160f2003a1e8f657924b5bc00191bb0ce9fa03/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/8e160f2003a1e8f657924b5bc00191bb0ce9fa03/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/8e160f2003a1e8f657924b5bc00191bb0ce9fa03/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [BuildingWareHouse.java#L38](https://github.com/Minecolonies/minecolonies/blob/8e160f2003a1e8f657924b5bc00191bb0ce9fa03/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingWareHouse.java#L38): Override the "equals" method in this class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2160)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 40 issues
* ![BLOCKER][BLOCKER] 3 blocker
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 34 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [Structure.java#L137](https://github.com/Minecolonies/minecolonies/blob/f7ab66886361c177bc34c87d4e191549feab0145/src/main/java/com/minecolonies/structures/helpers/Structure.java#L137): Close this "InputStream" in a "finally" clause. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2095)
1. ![BLOCKER][BLOCKER] [Structure.java#L247](https://github.com/Minecolonies/minecolonies/blob/f7ab66886361c177bc34c87d4e191549feab0145/src/main/java/com/minecolonies/structures/helpers/Structure.java#L247): Close this "InputStream" in a "finally" clause. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2095)
1. ![BLOCKER][BLOCKER] [Structure.java#L260](https://github.com/Minecolonies/minecolonies/blob/f7ab66886361c177bc34c87d4e191549feab0145/src/main/java/com/minecolonies/structures/helpers/Structure.java#L260): Close this "InputStream" in a "finally" clause. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2095)
1. ![MAJOR][MAJOR] [Structure.java#L63](https://github.com/Minecolonies/minecolonies/blob/f7ab66886361c177bc34c87d4e191549feab0145/src/main/java/com/minecolonies/structures/helpers/Structure.java#L63): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 37 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/f7ab66886361c177bc34c87d4e191549feab0145/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/f7ab66886361c177bc34c87d4e191549feab0145/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/f7ab66886361c177bc34c87d4e191549feab0145/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/f7ab66886361c177bc34c87d4e191549feab0145/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/f7ab66886361c177bc34c87d4e191549feab0145/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [BuildingWareHouse.java#L38](https://github.com/Minecolonies/minecolonies/blob/f7ab66886361c177bc34c87d4e191549feab0145/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingWareHouse.java#L38): Override the "equals" method in this class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2160)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Raycoms** [comment]
Just a port, working great, I tested it already.

**Pull request merged**